### PR TITLE
Side drawer for web and mobile

### PR DESCRIPTION
**Pull request by swanka**

### CODE REVIEW

**Review by joshgachnang — COMMENTED** [review]
Nice this looks great, works well on all 3! Some minor suggestions for stories would be adding buttons to test out the drawerTypes and both left/right. And maybe adding a scrollview to the story inside the drawer. That's the only UI component I can imagine giving us trouble in the drawer.

**Comment by joshgachnang** [comment]
Do we want the back button to close the drawer on Android? If so, can you add a TODO and we'll come back later?

**Comment by swanka** [comment]
> Do we want the back button to close the drawer on Android? If so, can you add a TODO and we'll come back later?

You should just be able to swipe it open/closed. Were you not able to close the drawer on Android?

File: packages/ui/package.json [hunk 1]
```diff
@@ -112,6 +112,10 @@
       "jest-watch-typeahead/testname"
     ]
   },
+  "dependencies": {
```
[the remark this answers is not in the document]
Comment (swanka) [reply packages/ui/package.json:115]: I can make them peer dependencies. 

Reanimated v3 seems to work ok? I haven't seen any errors come up

[the remark this answers is not in the document]
Comment (swanka) [reply packages/ui/package.json:115]: It looks like Expo 49 handles Reanimated v3. This issue was closed last month https://github.com/expo/expo/issues/21483

**Comment by joshgachnang** [comment]
> > Do we want the back button to close the drawer on Android? If so, can you add a TODO and we'll come back later?
> 
> You should just be able to swipe it open/closed. Were you not able to close the drawer on Android?

Yup! I could close it the normal way. But hitting the back button pulled me out of the app. I don't use Android enough to know if that's expected or I should expect the drawer to go away.

File: packages/ui/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshgachnang) [reply packages/ui/package.json:115]: Nice!

**Comment by swanka** [comment]
> > > Do we want the back button to close the drawer on Android? If so, can you add a TODO and we'll come back later?
> > 
> > 
> > You should just be able to swipe it open/closed. Were you not able to close the drawer on Android?
> 
> Yup! I could close it the normal way. But hitting the back button pulled me out of the app. I don't use Android enough to know if that's expected or I should expect the drawer to go away.

Oh interesting! I didn't think to use that button but I'm sure people might try. I just tested with the Sleeper app (for fantasy teams if you haven't used it) and they do allow the back button to close the drawer. I'll add that TODO

**Review by joshgachnang — APPROVED** [review]
Nice, this is great!

**Pull request merged**